### DISCUSSION TITLE
passed rate is not right

### DISCUSSION THREAD

**Discussion by zjd2626** [allure-framework/allure2#1348] [Bug Reports] [answered]
In my opinion,  when counting the test passed rate, skip testcases should be excluded.
for example, in some region, my testcases should not be executed so I mark skipped. but It cause my passed rate too low.
If I want to change passed rate, how can i do it.

**Comment by Avaddon-tfa** [comment]
Hi there! Well, what you say actually looks like a bug.
Do you mind creating an issue for us to fix that?

**Reply by zjd2626** [reply]
Hi, thanks for you reply. I think about it again. Maybe some scenarios is that some testcases mark as skip because some feature have issue or haven't been implemented,in this case, when counting test results, you may want to include skip testcases in all executed cases. Another scenario is that it does not need to be executed because conditions is not meet. In this case, when counting the test results, we hope to exclude the skip case from all executed cases. Is it possible to add a parameter to pytest.skip to support these two situations?

**Reply by zjd2626** [reply]
https://github.com/pytest-dev/pytest/discussions/8912

**Reply by Avaddon-tfa** [reply]
Thank you for the detailed explanation.

I believe we don't need to create an issue in the pytest repo. The truth is we can't change pytest success rate (SR) from the adapter site. But there are two solutions my colleagues propose:
1. Create an issue in allure2 (https://github.com/allure-framework/allure2) to discuss a high-level solution for your issue.
2. Filter out the tests to be skipped before the run with a webhook. Webhook is needed as pytest_collection_modifyitems is not implemented in the adapter.

**Comment by baev** [comment, marked as answer]
Fixed in [2.25.0](https://github.com/allure-framework/allure2/releases/tag/2.25.0)
